src/backend/base/langflow/components/models/AIMLModel.py

```diff
@@ -1,148 +1,128 @@
-    icon = "AI/ML"
-    chat_completion_url = "https://api.aimlapi.com/v1/chat/completions"
-
-    outputs = [
-        Output(display_name="Text", name="text_output", method="make_request"),
-    ]
+class AIMLModelComponent(LCModelComponent):
+    display_name = "AIML"
+    description = "Generates text using AIML LLMs."
+    icon = "AIML"
+    name = "AIMLModel"
+    documentation = "https://docs.aimlapi.com/api-reference"
 
-    inputs = [
-        DropdownInput(
-            name="model_name",
-            display_name="Model Name",
-            options=AIML_CHAT_MODELS,
-            required=True,
-        ),
-        SecretStrInput(
-            name="aiml_api_key",
-            display_name="AI/ML API Key",
-            value="AIML_API_KEY",
-        ),
-        MessageInput(name="input_value", display_name="Input", required=True),
+    inputs = LCModelComponent._base_inputs + [
         IntInput(
             name="max_tokens",
             display_name="Max Tokens",
             advanced=True,
             info="The maximum number of tokens to generate. Set to 0 for unlimited tokens.",
+            range_spec=RangeSpec(min=0, max=128000),
         ),
-        StrInput(
-            name="stop_tokens",
-            display_name="Stop Tokens",
-            info="Comma-separated list of tokens to signal the model to stop generating text.",
+        DictInput(name="model_kwargs", display_name="Model Kwargs", advanced=True),
+        BoolInput(
+            name="json_mode",
+            display_name="JSON Mode",
             advanced=True,
+            info="If True, it will output JSON regardless of passing a schema.",
         ),
-        IntInput(
-            name="top_k",
-            display_name="Top K",
-            info="Limits token selection to top K. (Default: 40)",
+        DictInput(
+            name="output_schema",
+            is_list=True,
+            display_name="Schema",
             advanced=True,
+            info="The schema for the Output of the model. You must pass the word JSON in the prompt. If left blank, JSON mode will be disabled.",
         ),
-        FloatInput(
-            name="top_p",
-            display_name="Top P",
-            info="Works together with top-k. (Default: 0.9)",
-            advanced=True,
+        DropdownInput(
+            name="model_name",
+            display_name="Model Name",
+            advanced=False,
+            options=AIML_CHAT_MODELS,
+            value=AIML_CHAT_MODELS[0],
         ),
-        FloatInput(
-            name="repeat_penalty",
-            display_name="Repeat Penalty",
-            info="Penalty for repetitions in generated text. (Default: 1.1)",
+        StrInput(
+            name="aiml_api_base",
+            display_name="AIML API Base",
             advanced=True,
+            info="The base URL of the OpenAI API. Defaults to https://api.aimlapi.com . You can change this to use other APIs like JinaChat, LocalAI e Prem.",
```

Comment:
I believe the intent of this component was to only hit the AI/ML API, so would lean towards removing this parameter. If users want to use a different base, we can add another component for that or use the generic OpenAI component. (I'm glad using the `base_url` here works using the aiml base - I tried with the `OpenAIEmbeddings` class and had issues).